tests/containers/bci_tests.pm

```diff
@@ -0,0 +1,135 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: bci-tests runner
+#   SUSE Linux Enterprise Base Container Images (SLE BCI)
+#   provides truly open, flexible and secure container images and application
+#   development tools for immediate use by developers and integrators without
+#   the lock-in imposed by alternative offerings.
+#   - install needed python environment
+#   - clone bci-tests repo
+#   - build project
+#   - run test for each given environment
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use Mojo::Base qw(consoletest);
+use XML::LibXML;
+use utils qw(systemctl zypper_call ensure_ca_certificates_suse_installed);
+use version_utils qw(get_os_release);
+use containers::common;
+use testapi;
+
+sub run {
+    my ($self) = @_;
+    $self->select_serial_terminal;
+
+    my $runtime        = get_required_var('RUNTIME');
+    my $bci_tests_repo = get_required_var('BCI_TESTS_REPO');
+    my $test_envs      = get_required_var('BCI_TEST_ENVS');
+    my $bci_devel_repo = get_var('BCI_DEVEL_REPO');
+    my $bci_timeout    = get_var('BCI_TIMEOUT', 900);
+
+    ensure_ca_certificates_suse_installed;
```

Comment:
from this line and till 50th - this code snippet which I have seen in several places . maybe not related to this PR but maybe we can extract it into function ?